### ISSUE TITLE
Publish the schemas in a human-readable form

### ISSUE THREAD

**Issue by blootsvoets** [RADAR-CNS/RADAR-Schemas#20]
We could do this in a GitHub.io page just posting the javadoc. There is a [avrodoc documentation generator](https://github.com/ept/avrodoc) but that doesn't support references to other types (Such as `restapi/app/application.avsc` refers to `restapi/app/server_status.avsc`.

**Comment by fnobilia** [comment]
Does it supports only embedded schemas?

**Comment by blootsvoets** [comment]
Yes: ept/avrodoc#19. And its unmaintained. I haven't found an alternative so far though...

**Comment by fnobilia** [comment]
At present, for the `Rest API` is almost impossible to avoid embedded schemas, we will end up with a single `.avsc` file containing almost all schemas. I don't think it can be easily maintained.

It is in my plan to structure the Rest API project to improve the modularisation and to simplify  future integrations. I may try to restructure the `Dataset` schema to mitigate its dependencies. If this is a low priority task and we can wait a little bit, I will get back to you with a more precise solution.
 
I see your point of publishing schemas in a human-readable form. I totally agree. However, are we sure we want to rely on a project that is unmaintained since 4 years? It is quite risky.

**Comment by blootsvoets** [comment]
No on the contrary, I think we shouldn't use Avrodoc. But we should have some form of documentation. Perhaps the route avsc -> JSON schema -> doc would also provide proper documentation, although we might loose some information in the conversion.

**Comment by blootsvoets** [comment]
BTW: by putting all files in a single protocol file, and putting them in an order where each type is defined before it is referenced, avrodoc seems to work fine.
![screen shot 2017-04-05 at 16 05 26](https://cloud.githubusercontent.com/assets/313450/24709269/a2e6e786-1a19-11e7-880b-bcd0d26fb4fb.png)
